File: ELIKAD_AlarmierungsApp/app/src/main/java/com/example/cora/elikad_alarmierungsapp/Data/Operation.java
package com.example.cora.elikad_alarmierungsapp.Data;

import java.time.DateTimeException;
import java.util.Date;

public class Operation {
    private int id;
    private String text;
    private String alarmlevel;
    private Date startDatetime;
    private Date endDatetime;
    private String caller;
    private Enum OperationType;
    private String controlcenterName;
    private Location location;

    public Operation(int id, String text, String alarmlevel, Date startDatetime, Date endDatetime, String caller, Enum operationType, String controlcenterName, Location location) {
        this.id = id;
        this.text = text;
        this.alarmlevel = alarmlevel;
        this.startDatetime = startDatetime;
        this.endDatetime = endDatetime;
        this.caller = caller;
        OperationType = operationType;
        this.controlcenterName = controlcenterName;
        this.location = location;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getAlarmlevel() {
        return alarmlevel;
    }

    public void setAlarmlevel(String alarmlevel) {
        this.alarmlevel = alarmlevel;
    }

    public Date getStartDatetime() {
        return startDatetime;
    }

    public void setStartDatetime(Date startDatetime) {
        this.startDatetime = startDatetime;
    }

    public Date getEndDatetime() {
        return endDatetime;
    }

    public void setEndDatetime(Date endDatetime) {
        this.endDatetime = endDatetime;
    }

    public String getCaller() {
        return caller;
    }

    public void setCaller(String caller) {
        this.caller = caller;
    }

    public Enum getOperationType() {
        return OperationType;
    }

    public void setOperationType(Enum operationType) {
        OperationType = operationType;
    }

    public String getControlcenterName() {
        return controlcenterName;
    }

    public void setControlcenterName(String controlcenterName) {
        this.controlcenterName = controlcenterName;
    }

    public Location getLocation() {
        return location;
    }

    public void setLocation(Location location) {
        this.location = location;
    }

    @Override
    public String toString() {
        return "Operation{" +
                "id=" + id +
                ", text='" + text + '\'' +
                ", alarmlevel='" + alarmlevel + '\'' +
                ", startDatetime=" + startDatetime +
                ", endDatetime=" + endDatetime +
                ", caller='" + caller + '\'' +
                ", OperationType=" + OperationType +
                ", controlcenterName='" + controlcenterName + '\'' +
                ", location=" + location +
                '}';
    }
}

File: ELIKAD_Verwaltungsclient/ELIKAD_Verwaltungsclient/Data/Operation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ELIKAD_Verwaltungsclient.Data
{
    public class Operation
    {
        public int Id { get; set; }
        public DateTime StartDatetime { get; set; }
        public DateTime? EndDatetime { get; set; }
        public string Text { get; set; }
        public string Type { get; set; }
        public string Alarmlevel { get; set; }
        public string ControlcenterName { get; set; }
        public int IdLocation { get; set; }

        public Operation(int id, DateTime startDatetime, DateTime endDatetime, string reason, string text, string alarmlevel, string controlcenterName, string type, int idLocation)
        {
            Id = id;
            StartDatetime = startDatetime;
            EndDatetime = endDatetime;
            Text = text;
            Alarmlevel = alarmlevel;
            ControlcenterName = controlcenterName;
            Type = type;
            IdLocation = IdLocation;
        }

        public Operation() { }
    }
}
